### PR TITLE
Feature/ISAL

### PR DESCRIPTION
**Pull request by willkoh**
In this PR, I added the ISA-L library to RDFS. The isal directory located at root contains all the files that wrap around the library. The code is borrowed from the actual hadoop repo. 

In tests/isal, there are two files, erasure_code_test.c and ErasureCodeTest.cc. The former is a test that was found in the hadoop repo that demonstrates how the wrapper should be used to encode and decode. The latter is a translation so that the test is similar to the other tests that we have. It's not really a test for anything other than ensuring that the library can be correctly loaded dynamically and that it can be used for our purposes.

Changes to CMakeLists, lint.sh, and vagrant_provision.sh has been made to allow for download/install of the library, as well as correctly building it into vagrant. main.sh was modified to include the new test.

Accidentally merged develop into the branch before feature/EC, so there are changes to files that are actually changes made into develop. Let me know if that's something that I should remove for the sake of workflow. Also, there are redundant commits because I'm a scrub when it comes to source control, so if some of these commits need to be cleaned up, let me know how to and I'll do them!

### CODE REVIEW

File: vagrant_provision.sh [hunk 1]
```diff
@@ -54,6 +56,14 @@ cp /home/vagrant/hadoop3/etc/hadoop/core-site.xml /home/vagrant/hadoop2/etc/hado
 cp /home/vagrant/hadoop3/etc/hadoop/hdfs-site.xml /home/vagrant/hadoop2/etc/hadoop/hdfs-site.xml
 
 
+# Setup Intel Storage Acceleration Library (ISA-L)
```
Comment (LINKIWI) [inline vagrant_provision.sh:59]: If these are required for compile, you'll need to add a script to `.ci/install` and invoke it in `.travis.yml` in `before_script`. See the existing files for examples.

How long does it take to download + compile the ISAL library? We should try to reduce build time impact, or at least mitigate impact with a build cache.

**Comment by jake-nyquist** [comment]
yeah ideally we won't have commits from develop in this pr

File: vagrant_provision.sh (hunk 1 above)
Comment (willkoh) [reply vagrant_provision.sh:59]: It's only about 20 mb so not that long, I think?

File: .ci/install/isal.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (LINKIWI) [inline .ci/install/isal.sh:1]: Make sure this script is exectuable, e.g.

```bash
$ chmod +x .ci/install/isal.sh
```

File: .ci/install/isal.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+set -ex
+
+ISAL_MIRROR=https://01.org/sites/default/files/downloads/intelr-storage-acceleration-library-open-source-version/isa-lopensrc2.13.tar.gz
```
Comment (LINKIWI) [inline .ci/install/isal.sh:5]: I mirrored this to a Rice server. Can you update the URL?

```
ISAL_MIRROR=http://kevinlin.web.rice.edu/static/isal.tar.gz
```

**Review by LINKIWI — APPROVED** [review]
approval for build step changes

**Comment by willkoh** [comment]
Going to make a new PR with cleaner commits and based off feature/EC !

**Pull request closed without merging**